driver-dse/src/main/java/com/datastax/driver/graph/AbstractGraphStatement.java

```diff
@@ -0,0 +1,141 @@
+     */
+
+    /**
+     * Get the current graph options configured for this statement.
+     *
+     * @return A payload map containing the configured Graph options.
+     */
+    public Map<String, ByteBuffer> getGraphOptions() {
+        return this.payload;
+    }
+
+    /**
+     * Set the Graph language to use for this statement.
+     *
+     * @param language
+     * @return This {@link com.datastax.driver.graph.AbstractGraphStatement} instance to allow chaining call.
+     */
+    public AbstractGraphStatement setGraphLanguage(String language) {
+        this.payload.put(GraphSession.GRAPH_LANGUAGE_KEY, ByteBuffer.wrap(language.getBytes()));
+        return this;
+    }
+
+    /**
+     * Set the Graph keyspace to use for this statement.
+     *
+     * @param graphKeyspace The keyspace to set. Throws an exception if the parameter is null or empty since this parameter is mandatory.
+     * @return This {@link com.datastax.driver.graph.AbstractGraphStatement} instance to allow chaining call.
+     */
+    public AbstractGraphStatement setGraphKeyspace(String graphKeyspace) {
+        if (graphKeyspace == null || graphKeyspace.isEmpty()) {
+            throw new InvalidQueryException("You cannot set null value or empty string to the keyspace for the Graph, this field is mandatory.");
+        }
+        this.payload.put(GraphSession.GRAPH_KEYSPACE_KEY, ByteBuffer.wrap(graphKeyspace.getBytes()));
+        return this;
+    }
+
+    /**
+     * Set the Graph traversal source to use for this statement.
+     *
+     * @param graphTraversalSource
+     * @return This {@link com.datastax.driver.graph.AbstractGraphStatement} instance to allow chaining call.
+     */
+    public AbstractGraphStatement setGraphSource(String graphTraversalSource) {
+        this.payload.put(GraphSession.GRAPH_SOURCE_KEY, ByteBuffer.wrap(graphTraversalSource.getBytes()));
+        return this;
+    }
+
+    /**
+     * Set the Graph rebinding name to use for this statement.
+     *
+     * @param graphRebinding
+     * @return This {@link com.datastax.driver.graph.AbstractGraphStatement} instance to allow chaining call.
+     */
+    public AbstractGraphStatement setGraphRebinding(String graphRebinding) {
+        this.payload.put(GraphSession.GRAPH_REBINDING_KEY, ByteBuffer.wrap(graphRebinding.getBytes()));
+        return this;
+    }
+
+    /**
+     * Set manually the custom payload to use for this statement.
+     * Please see {@link com.datastax.driver.core.Statement#setOutgoingPayload(java.util.Map)} for more
+     * information.
+     *
+     * @param payload The payload to set. Note that this will override the default values set in the {@link com.datastax.driver.graph.GraphSession}.
+     * @return This {@link com.datastax.driver.graph.AbstractGraphStatement} instance to allow chaining call.
+     */
+    public AbstractGraphStatement setOutgoingPayload(Map<String, ByteBuffer> payload) {
+        this.wrappedStatement.setOutgoingPayload(payload);
```

Comment:
Shouldn't this be: `this.payload.putAll(payload)` ? It seems that payloads are not consistently kept in sync between `his.payload` and `wrappedStatement`'s payload.